### PR TITLE
Updated CI

### PR DESCRIPTION
**Pull request by ohbus**
Removed duplicate maven caching and removed full namespace for the sonar plugin

Also updated the JDK to corretto

### CODE REVIEW

**Comment by ohbus** [comment]
@iluwatar I want to rejoin as a collaborator and help maintain this project and the other ones.

File: .github/workflows/maven-ci.yml [hunk 1]
```diff
@@ -51,17 +51,9 @@ jobs:
       uses: actions/setup-java@v3
       with:
         java-version: '17'
-        distribution: 'temurin'
+        distribution: 'corretto'
```
Comment (iluwatar) [inline .github/workflows/maven-ci.yml:54]: Hmm, why should we switch from temurin?

File: .github/workflows/maven-ci.yml [hunk 2]
```diff
@@ -51,17 +51,9 @@ jobs:
...
         cache: 'maven'
 
-    - name: Cache local Maven repository
-      uses: actions/cache@v3
-      with:
-        path: ~/.m2/repository
-        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-        restore-keys: |
-          ${{ runner.os }}-maven-
-
```
Comment (iluwatar) [inline .github/workflows/maven-ci.yml:64]: This seems to remove the local maven repository caching?

File: .github/workflows/maven-pr-builder.yml [hunk 3]
```diff
@@ -77,7 +69,15 @@ jobs:
...
         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-      run: xvfb-run ./mvnw clean verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.organization=iluwatar -Dsonar.projectKey=iluwatar_java-design-patterns -Dsonar.pullrequest.branch=$HEAD_REF -Dsonar.pullrequest.base=${{ github.base_ref }} -Dsonar.pullrequest.key=${{ github.event.pull_request.number }}
+      run: |
+        xvfb-run
+        ./mvnw clean verify
+        sonar:sonar
+        -Dsonar.pullrequest.branch=$HEAD_REF
+        -Dsonar.pullrequest.base=$PR_BASE_REF
+        -Dsonar.pullrequest.key=$PR_NUMBER
```
Comment (iluwatar) [inline .github/workflows/maven-pr-builder.yml:83]: Yeah, this looks prettier (if it works)

**Review by iluwatar — CHANGES_REQUESTED** [review]

**Comment by iluwatar** [comment]
> @iluwatar I want to rejoin as a collaborator and help maintain this project and the other ones.

Noted, let me consider

**Pull request closed without merging**